code/game/objects/structures/railing.dm

```diff
@@ -25,7 +25,8 @@
 	..()
 	if(climbable)
 		verbs += /obj/structure/proc/climb_on
-	update_icon(FALSE)
+	spawn(0)
```

Comment:
It probably doesn't work because it's in New and not in Initialize, putting it in initialize will make sure that when railing updates its icon, all neighbors exist. In fact this whole new can be just initialize, try changing
````
New()
...()
```
to 
```
Initialize()
. = ..()
```